### DISCUSSION TITLE
Add Google Firebase Analytics at least to Android and iOS implementations

### DISCUSSION THREAD

**Discussion by theromis** [godotengine/godot-proposals#10258] [Platforms]
### Describe the project you are working on

Need it and wrote simple plugin for android, but adding it to iOS too painful.

### Describe the problem or limitation you are having in your project

Adding Google Analytics to iOS too painful.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Just build engine with google analytics linked in.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Just report to google events what developers want to research.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Just write plugin/module.

### Is there a reason why this should be core and not an add-on in the asset library?

Building addon is pretty hard to build on iOS.

**Comment by Calinou** [comment]
Moving to discussion, as there is no concrete technical implementation provided.

**Comment by theromis** [comment]
https://github.com/funseek/godot-ios-firebase-analytics
here is ios firebase analytics for 3.5, code by itself pretty simple, can be easily converted to 4x
for android I have a plugin, can share it here
https://github.com/FeatureKillersGames/FirebaseAnalyticsGodotAndroidPlugin
this is plugin forked from in my android implementation

**Comment by Abab-bk** [comment]
IMO, Adding any SDK is the worst idea, Godot is just a game engine, and it should make server add plugin to Godot, rather Godot add plugin to server.

**Reply by theromis** [reply]
Agreed, extra dependencies is not ideal, but adding any sdk for iOS real pain, just look at this part:
```
To generate Godot headers you need to run compilation command inside godot submodule directory.
Example:

./scripts/generate_headers.sh
run pod install. [CocoaPods](https://cocoapods.org/)

pod install
Building a .a library

./scripts/release_static_library.sh 3.5
...
```
maybe making iOS modules as an extra library can help?
btw: Unity have Firebase Analytics out of the box, thought godot can do the same
